src/SDL/Font.hs

```diff
@@ -244,14 +244,14 @@ blended (Font font) (V4 r g b a) text =
 
 -- Analogous to Data.Text.Foreign.useAsPtr, just appends a null-byte.
 -- FIXME: Is this even necessary?
-withText :: Text -> (Ptr Word16 -> IO a) -> IO a
+withText :: Text -> (Ptr Word8 -> IO a) -> IO a
```

Comment:
That would be `CString` now.

Reply:
Thanks a lot for the tips. Do you think we could just use `Data.Text.Foreign.useAsPtr` or something similar instead?

Reply:
I have a similar issue with text-2 in `dear-imgui`. But I know from the code I need to append NUL in a few places here and there.

We need `Data.Text.Foreign.useAsCString` and there's https://github.com/haskell/text/pull/254 already.

Reply:
https://github.com/haskell/text/pull/431 - it's on the books now.

Would it be reasonable to request `text <2 || >= 2.1` and don't ship an own copy? :thinking:

Reply:
Amazing! Yes, procrastination works. :D

Reply:
Yes, I think it's reasonable to request text <2 || >= 2.1. I will do that myself. In `dear-imgui`, would that permit to change the condition in https://github.com/haskell-game/dear-imgui.hs/blob/main/src/DearImGui/Internal/Text.hs#L29-L50 to `MIN_VERSION_text(2,0,1)` and the first branch of the conditional to `withCString = Data.Text.Foreign.withCString` and then the second branch stays untouched?